homeassistant/components/mqtt/light/__init__.py

```diff
@@ -27,18 +23,25 @@
 
 def validate_mqtt_light(value):
     """Validate MQTT light schema."""
+    from . import schema_basic
+    from . import schema_json
+    from . import schema_template
+
     schemas = {
         'basic': schema_basic.PLATFORM_SCHEMA_BASIC,
         'json': schema_json.PLATFORM_SCHEMA_JSON,
         'template': schema_template.PLATFORM_SCHEMA_TEMPLATE,
     }
     return schemas[value[CONF_SCHEMA]](value)
 
-
-PLATFORM_SCHEMA = vol.All(vol.Schema({
+SCHEMA_SCHEMA = {
     vol.Optional(CONF_SCHEMA, default='basic'): vol.All(
         vol.Lower, vol.Any('basic', 'json', 'template'))
-}, extra=vol.ALLOW_EXTRA), validate_mqtt_light)
+}
+
+PLATFORM_SCHEMA = vol.All(vol.Schema(
+    SCHEMA_SCHEMA,
```

Comment:
Does this work? shouldn't we just extend and then set vol.ALLOW_EXTRA?

Reply:
Yeah, works, but let's leave it for a later MQTT PR.